libs-haskell/test-utils/DA/Test/Calling.hs

```diff
@@ -0,0 +1,69 @@
+-- Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module DA.Test.Calling
+  ( runCreateProcessForStdout
+  , callCommandForStdout
+  , runCreateProcessQuiet
+  , callProcessQuiet
+  , callCommandQuiet
+  , waitForProcess'
+  , withDevNull
+  , P.callCommand
+  ) where
+
+import Control.Exception (Exception,throwIO)
+import Control.Monad (unless)
+import Data.Typeable (Typeable)
+import System.Exit (ExitCode(ExitSuccess))
+import System.IO.Extra (Handle,hPutStr,stderr,withTempFile,withFile,IOMode(WriteMode))
+import System.Process (CreateProcess(..),ProcessHandle,shell,proc)
+import qualified System.Process as P
+
+-- | Like call process but returning stdout.
+runCreateProcessForStdout :: CreateProcess -> IO String
+runCreateProcessForStdout createProcess = do
+    -- We use `repeat ' '` to keep stdin open. Really we would just
+    -- like to inherit stdin but readCreateProcessWithExitCode does
+    -- not allow us to overwrite just that and I don’t want to
+    -- reimplement everything.
+    (exit, out, err) <- P.readCreateProcessWithExitCode createProcess (repeat ' ')
+    hPutStr stderr err
+    unless (exit == ExitSuccess) $ throwIO $ ProcessExitFailure exit createProcess
+    return out
+
+callCommandForStdout :: String -> IO String
```

Comment:
The whole `runProcessQuiet` stuff already got moved here. You need to rebase.

Reply:
Actually, the `callProcessSilent` in DA.Test.Util was not being used by the code I was refactoring (`IntegrationTests.hs`), which contained an almost duplication of the silent/quiet thing. I merged these together now.